src/modules/logicModule/LM_PC_FundingPot_v1.sol

```diff
@@ -272,53 +355,243 @@ contract LM_PC_FundingPot_v1 is
+                accessCriteriaId_ == uint8(AccessCriteriaType.NFT)
+                    && nftContract_ == address(0)
             )
                 || (
-                    accessCriteria_.accessCriteriaType == AccessCriteriaType.MERKLE
-                        && accessCriteria_.merkleRoot == bytes32("")
+                    accessCriteriaId_ == uint8(AccessCriteriaType.MERKLE)
+                        && merkleRoot_ == bytes32("")
                 )
                 || (
-                    accessCriteria_.accessCriteriaType == AccessCriteriaType.LIST
-                        && accessCriteria_.allowedAddresses.length == 0
+                    accessCriteriaId_ == uint8(AccessCriteriaType.LIST)
+                        && allowedAddresses_.length == 0
                 )
         ) {
             revert Module__LM_PC_FundingPot__MissingRequiredAccessCriteriaData();
         }
-        uint8 accessCriteriaId = roundIdtoAccessId[roundId_];
-        round.accessCriterias[accessCriteriaId] = accessCriteria_;
 
-        emit AccessCriteriaSet(roundId_, accessCriteriaId, accessCriteria_);
+        AccessCriteriaType accessCriteriaType =
+            AccessCriteriaType(accessCriteriaId_);
+        round.accessCriterias[accessCriteriaId_].accessCriteriaType =
+            accessCriteriaType;
+        round.accessCriterias[accessCriteriaId_].nftContract = nftContract_;
+        round.accessCriterias[accessCriteriaId_].merkleRoot = merkleRoot_;
 
-        roundIdtoAccessId[roundId_] += 1;
+        for (uint i = 0; i < allowedAddresses_.length; i++) {
+            round.accessCriterias[accessCriteriaId_].allowedAddresses[allowedAddresses_[i]]
+            = true;
+        }
+
+        emit AccessCriteriaSet(roundId_, accessCriteriaId_);
     }
 
     /// @inheritdoc ILM_PC_FundingPot_v1
     function editAccessCriteriaForRound(
         uint64 roundId_,
         uint8 accessCriteriaId_,
-        AccessCriteria memory accessCriteria_
+        address nftContract_,
+        bytes32 merkleRoot_,
+        address[] calldata allowedAddresses_
     ) external onlyModuleRole(FUNDING_POT_ADMIN_ROLE) {
-        if (accessCriteriaId_ >= roundIdtoAccessId[roundId_]) {
+        Round storage round = rounds[roundId_];
+        if (accessCriteriaId_ > MAX_ACCESS_CRITERIA_ID) {
             revert Module__LM_PC_FundingPot__InvalidAccessCriteriaId();
         }
+
+        _validateEditRoundParameters(round);
+
+        AccessCriteriaType accessCriteriaType =
+            AccessCriteriaType(accessCriteriaId_);
+        round.accessCriterias[accessCriteriaId_].accessCriteriaType =
+            accessCriteriaType;
+        round.accessCriterias[accessCriteriaId_].nftContract = nftContract_;
+        round.accessCriterias[accessCriteriaId_].merkleRoot = merkleRoot_;
+
+        for (uint i = 0; i < allowedAddresses_.length; i++) {
+            round.accessCriterias[accessCriteriaId_].allowedAddresses[allowedAddresses_[i]]
+            = true;
+        }
```

Comment:
Right now we can only add to the list of allowed addresses, I think we should also be able to remove addresses. Maybe just add another function that takes a list of `permittedAddresses` and sets the bools to false in the mapping for these addresses?